backend/editor/entries.py

```diff
@@ -101,4 +186,40 @@ def update_nodes(label, entry, incomingData):
 
     params = dict(incomingData, id=entry)
     result = session.run(" ".join(query), params)
-    return result
\ No newline at end of file
+    return result
+
+def update_node_children(entry, incomingData):
+    """
+    Helper function used for updation of node children with given id 
+    """
+    # Parse node ids from Neo4j Record object
+    current_children = [record["b.id"] for record in list(get_children(entry))]
+    deleted_children = list(set(current_children) ^ set(incomingData))

```

Comment:
This is not clear, you are mixing deleted_children and added_children.

So really you should have this instead:
```suggestion
    deleted_children = set(current_children) - set(incomingData)
    added_children = set(incomingData) - set(current_children)
```
Then your code can treat them separately.